### PR TITLE
Add broadcast importance (fixes #669)

### LINKED ISSUE #669: Broadcast timer overwrites other broadcasts
Currently the broadcast timer overwrites other broadcasts. It would be nice if the broadcast timer doesn't get send while another broadcast is displayed.

### PR DESCRIPTION
**Pull request by ChillerDragon**
The broadcast timer is marked as unimportant and it won't overwrite if there was another broadcast sent less than 6 seconds ago. So vanilla clients won't miss any important news c:

### CODE REVIEW

**Comment by Learath2** [comment]
Could you implement this with an importance level instead of a boolean? e.g. broadcast with highest importance is shown.

I'd be okay with preset levels too e.g.
```
enum {
    BROADCAST_LOW = 0,
    BROADCAST_MID,
    BROADCAST_HIGH
}
```

**Comment by heinrich5991** [comment]
Wait a sec. Maybe this should rather append broadcasts instead of replacing them – i.e. appending different broadcasts.

+1 for using a branch, @ChillerDragon :)

**Comment by heinrich5991** [comment]
If not: That config variable can probably be omitted and replaced by the number of seconds the client displays a broadcast.

**Comment by ChillerDragon** [comment]
@Learath2 i thought about using importance levels like in DDNet++ but DDNet++ is a blown source with a lot of broadcasts. I tought its is better to keep DDNet source clean. We can later on easy upgrade it to a multi importance system. But for now there is just the timer making trouble as far as i know.

If we want to prevent helpers or moderators from overwriting map releases or admin broadcasts later on we can easy upgrade the system. But i would keep the bool until someone actually wants and needs more importance levels.


@heinrich5991 when i implemented this in DDNet++ i also thought about appending but it became a bit ugly. But if we only use it for the timer we know the exact length and it could work somehow. But it can still confuse. I think in the timer case we can ignore time for 6 seconds. But if you really think that we should append i can try working on that again.

I am not sure if i actually checked the client time back then when i first built this thing. But 6 seconds seem to work pretty good. I just implemented the config because i was told so in the dev discord and i like configs c:

File: src/engine/shared/config_variables.h [hunk 1]
```diff
@@ -187,6 +187,7 @@ MACRO_CONFIG_INT(SvHit, sv_hit, 1, 0, 1, CFGFLAG_SERVER|CFGFLAG_GAME, "Whether p
 MACRO_CONFIG_INT(SvEndlessDrag, sv_endless_drag, 0, 0, 1, CFGFLAG_SERVER|CFGFLAG_GAME, "Turns endless hooking on/off")
 MACRO_CONFIG_INT(SvTestingCommands, sv_test_cmds, 0, 0, 1, CFGFLAG_SERVER, "Turns testing commands aka cheats on/off")
 MACRO_CONFIG_INT(SvFreezeDelay, sv_freeze_delay, 3, 1, 30, CFGFLAG_SERVER|CFGFLAG_GAME, "How many seconds the players will remain frozen (applies to all except delayed freeze in switch layer & deepfreeze)")
+MACRO_CONFIG_INT(SvBroadcastDelay, sv_broadcast_delay, 6, 1, 180, CFGFLAG_SERVER, "How many seconds unimportant broadcasts won't overwrite important ones")
```
Comment (heinrich5991) [inline src/engine/shared/config_variables.h]: Okay, I think you can just make this 10 seconds and omit the config variable.

File: src/game/server/gamecontext.cpp [hunk 2]
```diff
@@ -398,11 +398,35 @@ void CGameContext::SendWeaponPickup(int ClientID, int Weapon)
...
+
+		for(int i = 0; i < MAX_CLIENTS; i++)
+		{
+			if(m_apPlayers[i])
+			{
+				m_apPlayers[i]->m_LastBroadcastImportance = true;
+				m_apPlayers[i]->m_LastBroadcast = Server()->Tick();
+			}
+		}
+		return;
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp]: This seems to need handling for `IsImportant = false` and `ClientID = -1`. Such a handling could consist of a `dbg_assert(IsImportant, "broadcast messages to all players must be important");`.

**Comment by def-** [comment]
Thank you!

**Pull request merged**